### PR TITLE
[PROTOCOL-87] Integration Tests: Create Repayment Scenarios I

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Integration tests to cover the loan repayments.

It includes:
- Two integration tests:
-- Repay in 3 times with loan terms: TokenName/ETH ; 30 days; 1000 TokenName
-- Repay in 4 times with loan terms: TokenName/LINK ; 60 days; 800 TokenName

Screen Shot 2020-07-27 at 7 08 25 PM

### CODE REVIEW

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  
+  // Take out a loan.
+  console.log(`Taking out loan id ${lastLoanID}...`);
+  const takeOutLoanResult = await loansInstance.takeOutLoan(lastLoanID, amountWei, borrowerTxConfig);
+  loans
+    .loanTakenOut(takeOutLoanResult)
+    .emitted(lastLoanID, borrowerTxConfig.from, amountWei);
+
+  // Set a lower price for Token/ETH.
+  console.log(`Settings final (lower) oracle price: 1 ${tokenName} = ${finalOraclePrice.toFixed(0)} WEI = ${toUnits(finalOraclePrice, 18)} ETHER`);
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:115]: I think this new oracle price is not needed in the scenario.

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+  // Advance time to make first payment 10 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const paymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(paymentTimestamp);
+
+  // Make 1st payment
+  console.log(`Repaying loan id ${lastLoanID}...`);
+  console.log('Making 1st payment...');
+  await token.approve(loansInstance.address, payment, {from:borrower});
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:131]: I  think you don't need this approve call (to loansInstance).

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Advance time to make first payment 20 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const secondPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(secondPaymentTimestamp);
+
+  // Make 2nd payment
+  console.log('Making 2nd payment...');
+  await token.approve(loansInstance.address, payment, {from:borrower});
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:142]: The same here

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
+
+  // Advance time to make first payment 29 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const thirdPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(9));
+  await timer.advanceBlockAtTime(thirdPaymentTimestamp);
+
+  // Make 3rd payment
+  console.log('Making 3rd payment...');
+  await token.approve(loansInstance.address, payment, {from:borrower});
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:153]: The same here

File: test-integration/repay-loan/1_repay_loan_3_payments.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  // Advance time to make first payment 20 days later 
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const secondPaymentTimestamp = await timer.getCurrentTimestampInSecondsAndSum(daysToSeconds(10));
+  await timer.advanceBlockAtTime(secondPaymentTimestamp);
+
+  // Make 2nd payment
+  console.log('Making 2nd payment...');
+  await token.approve(loansInstance.address, payment, {from:borrower});
+  await token.approve(lendingPoolInstance.address, payment, {from:borrower});
+  await loansInstance.repay(payment, lastLoanID, borrowerTxConfig);
```
Comment (salazarguille) [inline test-integration/repay-loan/1_repay_loan_3_payments.js:144]: Please add assertions for the Repayment events

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 1 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:115]: Removed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 2 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:131]: Removed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 3 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:142]: Removed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 4 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:153]: Removed

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 5 above)
Comment (sypeer) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:144]: Added

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 1 above)
Comment (salazarguille) [reply test-integration/repay-loan/1_repay_loan_3_payments.js]: I think it wasn't removed.

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 2 above)
Comment (salazarguille) [reply test-integration/repay-loan/1_repay_loan_3_payments.js]: The same. I think it wasn't removed.

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 1 above)
Comment (salazarguille) [reply test-integration/repay-loan/1_repay_loan_3_payments.js]: I think it wasn't removed. https://github.com/teller-protocol/teller-protocol-v1/blob/7dbb5ccbfaeb02ead9963a336117aa81f0869b70/test-integration/repay-loan/1_repay_loan_3_payments.js#L116

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 3 above)
Comment (salazarguille) [reply test-integration/repay-loan/1_repay_loan_3_payments.js]: The same

File: test-integration/repay-loan/1_repay_loan_3_payments.js (hunk 5 above)
Comment (salazarguille) [reply test-integration/repay-loan/1_repay_loan_3_payments.js:145]: I can't see the assertion.

**Pull request closed without merging**